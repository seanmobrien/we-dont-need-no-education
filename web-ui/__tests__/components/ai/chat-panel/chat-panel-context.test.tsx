import React from 'react';
import { render } from '@/__tests__/test-utils';
import {
  ChatPanelProvider,
  useChatPanelContext,
} from '@/components/ai/chat-panel/chat-panel-context';

// Test component to access context
const TestComponent = () => {
  const context = useChatPanelContext();
  return (
    <div data-testid="test-component">
      <span data-testid="is-docked">{context.isDocked.toString()}</span>
      <span data-testid="is-floating">{context.isFloating.toString()}</span>
      <span data-testid="is-inline">{context.isInline.toString()}</span>
      <span data-testid="position">{context.config.position}</span>
    </div>
  );
};

describe('ChatPanelContext', () => {
  it('renders with default state snapshot', () => {
    const { container } = render(
      <ChatPanelProvider>
        <TestComponent />
      </ChatPanelProvider>,
    );
    expect(container).toMatchSnapshot();
  }, 10000);

  it('provides default context values', () => {
    const { getByTestId } = render(
      <ChatPanelProvider>
        <TestComponent />
      </ChatPanelProvider>,
    );

    expect(getByTestId('is-docked')).toHaveTextContent('false');
    expect(getByTestId('is-floating')).toHaveTextContent('false');
    expect(getByTestId('is-inline')).toHaveTextContent('true');
    expect(getByTestId('position')).toHaveTextContent('inline');
  }, 10000);
});
