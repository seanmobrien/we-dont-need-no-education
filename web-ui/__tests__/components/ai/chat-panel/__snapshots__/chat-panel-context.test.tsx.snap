// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ChatPanelContext renders with default state snapshot 1`] = `
<div>
  <div
    data-testid="test-component"
  >
    <span
      data-testid="is-docked"
    >
      false
    </span>
    <span
      data-testid="is-floating"
    >
      false
    </span>
    <span
      data-testid="is-inline"
    >
      true
    </span>
    <span
      data-testid="position"
    >
      inline
    </span>
  </div>
</div>
`;
